### DISCUSSION TITLE
Web Extension Notebook Renderers

### DISCUSSION THREAD

**Discussion by cedricbuild** [microsoft/vscode-discussions#62] [Extension Development QnA] [answered]
Web extension have limited functionality as they run in the browser. 

Is it possible to know extension runtime environment in notebook renderer?

**Comment by alefragnani** [comment, marked as answer]
Hi @cedricbuild ,

You may be able to use `vscode.env.appHost`, which by documentation says:

> /**
         * The hosted location of the application
         * On desktop this is 'desktop'
         * In the web this is the specified embedder i.e. 'github.dev', 'codespaces', or 'web' if the embedder
         * does not provide that information
         */

Hope this helps

**Reply by cedricbuild** [reply]
does this work in notebook renderer?

**Reply by alefragnani** [reply]
I'm not sure. Never developed a notebook renderer, so I don't know its architecture. 

But if you can do import `vscode` definitions, you will have it available

```ts
import { env } from "vscode";
...
const runningOn = env.AppHost;
console.log(runningOn);
```

**Reply by RandomFractals** [reply]
@cedricbuild you can see what's involved in making vscode notebook renderers extension work in vscode web editor on vscode.dev, github.dev and code spaces in this ticket for data table renderers vscode extension: https://github.com/RandomFractals/vscode-data-table/issues/122

That extension is no longer in the marketplace, as I unpublished it due to low number of installs back in April, but you can see the code changes and steps to make yours work in vscode web editor with code notebooks in that ticket.
